src/Hadolint/Rules.hs

```diff
@@ -1164,3 +1166,21 @@ noIllegalInstructionInOnbuild = instructionRule code severity message check
     check (OnBuild (From _)) = False
     check (OnBuild (Maintainer _)) = False
     check _ = True
+
+relativeCopyWithoutWorkdir :: Rule
+relativeCopyWithoutWorkdir = instructionRuleState code severity message check ("", [])
+  where
+    code = "DL3044"
+    severity = WarningC
+    message = "`COPY` to a relative destination without `WORKDIR` set."
+    check st _ (From BaseImage {image, alias})
```

Comment:
Try using a bit more of pattern matching: `(From BaseImage {image, alias = Just theAlias}`

Reply:
I did, but I am not sure this is any better than before. But its not any worse, so why not.